### ISSUE TITLE
Clouddriver unable to use AMI shared with other AWS account

### ISSUE THREAD

**Issue by dgreene-r7** [spinnaker/spinnaker#3106]
### Issue Summary:
Unable to spin up an ASG from a shared AMI when the account the deployment is happening in and the account that shares the AMI are not the same account in which Spinnaker is deployed.

### Cloud Provider(s):
AWS

### Environment:
Spinnaker deployed in account A, deploying ASG in account B from an AMI shared with account B from account C.

### Feature Area: 
Clouddriver

### Description:
When Spinnaker is set up in Account A and tries to deploy an ASG in account B (IAM role configured correctly) with an AMI shared from account C, an `IllegalArgumentException` occurs with the following message: [`Unable to find owner of resolved AMI...`](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AllowLaunchAtomicOperation.groovy#L102).

The text of the error in the UI (with account specific stuff redacted):
```
Exception ( Monitor Deploy )
Unable to find owner of resolved AMI com.netflix.spinnaker.clouddriver.aws.deploy.ResolvedAmiResult(ami-AMI_ID, us-east-1, ami-AMI_ID, hvm, ACCOUNT_NUMBER, [{DeviceName: /dev/sda1,Ebs: {Encrypted: false,DeleteOnTermination: true,SnapshotId: snap-SNAPSHOT_ID,VolumeSize: 8,VolumeType: gp2},}, {DeviceName: /dev/sdb,VirtualName: ephemeral0,}, {DeviceName: /dev/sdc,VirtualName: ephemeral1,}], false)
```

### Steps to Reproduce:
See Description

### Additional Details:
Clouddriver v2.47.0

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by dgreene-r7** [comment]
@spinnakerbot remove-label stale

**Comment by adamlazz** [comment]
This issue seems similar to https://github.com/spinnaker/spinnaker/issues/3355

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.

**Comment by dsilvasc** [comment]
@spinnakerbot remove-label stale

**Comment by costimuraru** [comment]
We're seeing the same issue with 1.16.0.

We're using Account A to bake the image and we're sharing the AMI with Account B.

`Find image from Tags` correctly retrieves the right AMI for Account B. However, the Deployment fails after 5 seconds:
```
Unable to find owner of resolved AMI com.netflix.spinnaker.clouddriver.aws.deploy.ResolvedAmiResult(ami-00d58a7de88b99185, ap-northeast-1, ami-00d58a7de88b99185, hvm, 881911269000, [{DeviceName: /dev/sda1,Ebs: {DeleteOnTermination: false,SnapshotId: snap-015d3ee3501ae4333,VolumeSize: 8,VolumeType: standard,Encrypted: false,},}], false)
```

Where 881911269000 is Account A.

**Comment by costimuraru** [comment]
<img width="1078" alt="Screenshot 2019-09-11 04 45 06" src="https://user-images.githubusercontent.com/952836/64661834-f08fbc00-d44e-11e9-8403-320f94ac003e.png">

**Comment by costimuraru** [comment]
Figured out the problem. We were baking the image in Account A, but that specific account was not added in the list of accounts for the aws provider, in spinnaker. When spinnaker was trying to deploy the image in Account B, it was trying to fetch the credentials for Account A, which were inexistent, hence the error.
We added Account A to the list of aws accouts (in clouddriver, via hal) and it works fine now.
